brownie/runlogs/2022_11_strategist.py

```diff
@@ -0,0 +1,301 @@
+# --------------------------------
+# Nov 3, 2022 OGV Buyback
+# 
+
+
+from world import *
+from ape_safe import ApeSafe
+
+
+MIN_PERCENT_AFTER_SLIPPAGE = 0.97
+REWARDS = "0x7d82E86CF1496f9485a8ea04012afeb3C7489397"
+buyback = Contract.from_abi("Buyback", vault_core.trusteeAddress(), buyback.abi)
+ogv = Contract.from_explorer("0x9c354503C38481a7A7a51629142963F98eCC12D0")
+
+BUYBACK_AMOUNT = ousd.balanceOf(buyback)
+
+
+amount_no_mev = 0
+
+# Nominal price sim
+with TemporaryFork():
+    before = ogv.balanceOf(REWARDS)
+    buyback.swapNow(1e18, 1, {'from': STRATEGIST})
+    after = ogv.balanceOf(REWARDS)
+    amount_no_mev = after - before
+
+print("Best-case buyback amount %s OUSD for %s OGV" % (
+    c18(BUYBACK_AMOUNT),
+    c18(amount_no_mev*BUYBACK_AMOUNT/int(1e18))
+))
+print("Best-case price $%f"%(1e18 / amount_no_mev))
+
+
+# Expected Swap sim
+with TemporaryFork():
+    before = ogv.balanceOf(REWARDS)
+    buyback.swapNow(BUYBACK_AMOUNT, 1, {'from': STRATEGIST})
+    print(history[-1].call_trace(True))
+    after = ogv.balanceOf(REWARDS)
+    amount_no_mev = after - before
+
+print("Target buyback amount %s OUSD for %s OGV"%(c18(BUYBACK_AMOUNT), c18(amount_no_mev)))
+print("Target price $%f"%(BUYBACK_AMOUNT / amount_no_mev))
+print("Min amount %s OGV"%(c18(int(amount_no_mev*MIN_PERCENT_AFTER_SLIPPAGE))))
+print("Submit with 500,000 gas")
+
+# Actual Swap TX
+txs = [
+    buyback.swapNow(BUYBACK_AMOUNT, int(amount_no_mev*MIN_PERCENT_AFTER_SLIPPAGE), {'from': STRATEGIST})
+]
+
+# Send
+safe = ApeSafe('0xF14BBdf064E3F67f51cd9BD646aE3716aD938FDC')
+safe_tx = safe.multisend_from_receipts(txs)
+safe.sign_with_frame(safe_tx)
+r = safe.post_transaction(safe_tx)
+
+
+
+
+# --------------------------------
+# Nov 8, 2022 Initial MetaStrategy Depost
+# 
+
+from world import *
+from allocations import *
+from ape_safe import ApeSafe
+
+with TemporaryFork():
+    print(load_from_blockchain())
+    before = vault_core.totalValue()
+    txs = [
+        vault_value_checker.takeSnapshot({"from": STRATEGIST}),
+        vault_admin.reallocate(OUSD_META_STRAT, COMP_STRAT, [usdc], [int(481000 * 1e6)], {'from': STRATEGIST}),
+        vault_value_checker.checkLoss(-480800*1e18, {"from": STRATEGIST}),
```

Comment:
shouldn't this be a positive number?

Reply:
The vault vaule is going up because the Metastrategy is printing OUSD, which increases the vault value. A value value increase is checked as a "negative loss".